torch/distributed/tensor/_random.py

```diff
@@ -170,7 +170,11 @@ def __init__(
         super().__init__(_resolve_device(device_mesh=device_mesh))
         assert self._device_handle is not None
         # DTensor RNG tracker so far only supports CUDA/CUDA-like devices
-        if self._device.type != "cuda":
+        if self._device.type != "cuda" and (
+            self._device.type == "cpu"
+            or not hasattr(self._device_handle, "set_rng_state")
+            or not hasattr(self._device_handle, "get_rng_state")
+        ):
```

Comment:
Oh, wait a minute. Why do we still have `if self._device.type != "cuda" and` here?

Reply:
Yeah, I used to think that `cuda` should be allowed anyway, but removing it directly would be ok as `torch.cuda` both has `set_rng_state()` and `get_rng_state()`

Reply:
Have made changes and please take a look, thanks!